File: AngularApp/src/app/__mocks__/mock.band.ts
import { Band } from '../model';
import { faker } from '@faker-js/faker';
import { MockMusic } from './mock.music';

export class MockBand {
  private static _instance: MockBand;
  private faker: typeof faker;

  private constructor() {
    this.faker = faker;
  }

  public static instance(): MockBand {
    if (!MockBand._instance) {
      MockBand._instance = new MockBand();
    }
    return MockBand._instance;
  }

  public getFaker(): Band {
    const mockBand: Band = ({
      id: faker.string.uuid(),
      name: faker.music.songName(),
      description: faker.lorem.words(),
      backdrop: faker.internet.url(),
      albums: Array.from({ length: 5 }, () => ({
        id: faker.string.uuid(),
        name: faker.music.songName(),
        backdrop: faker.internet.url(),
        flatId: faker.string.uuid(),
        bandId: faker.string.uuid(),
        musics: MockMusic.instance().generateMusicList(2)
      })),
    });
    return mockBand;
  }

  public generateBandList(count: number): Band[] {
    const BandList: Band[] = [];
    for (let i = 0; i < count; i++) {
      const Band: Band = this.getFaker();
      BandList.push(Band);
    }
    return BandList;
  }
}

File: LiteStreaming.XUnitTest/__mock__/MockBand.cs
using Application.Streaming.Dto;
using Domain.Streaming.Agreggates;
using Bogus;

namespace __mock__;
public class MockBand
{
    private static readonly Lazy<MockBand> _instance = new Lazy<MockBand>(() => new MockBand());

    public static MockBand Instance => _instance.Value;

    private MockBand() { }

    public Band GetFaker()
    {
        var mockMusics = MockMusic.Instance.GetListFaker(5);
        var mockAlbum = MockAlbum.Instance.GetFaker(mockMusics);

        var fakeBand = new Faker<Band>()
            .RuleFor(b => b.Id, f => f.Random.Guid())
            .RuleFor(b => b.Name, f => f.Name.FullName())
            .RuleFor(b => b.Description, f => f.Lorem.Sentence())
            .RuleFor(b => b.Backdrop, f => f.Image.PlaceImgUrl())
            .RuleFor(b => b.Albums, f => new List<Album> { mockAlbum })
            .Generate();

        return fakeBand;
    }

    public BandDto GetDtoFromBand(Band band)
    {
        var fakeBandDto = new Faker<BandDto>()
            .RuleFor(b => b.Id, f => band.Id)
            .RuleFor(b => b.Name, f => band.Name)
            .RuleFor(b => b.Description, f => band.Description)
            .RuleFor(b => b.Backdrop, f => band.Backdrop)
            .RuleFor(b => b.Albums, f => MockAlbum.Instance.GetDtoListFromAlbumList(band.Albums))
            .Generate();

        return fakeBandDto;
    }
    public List<Band> GetListFaker(int count)
    {
        var bandList = new List<Band>();
        for (var i = 0; i < count; i++)
        {
            bandList.Add(GetFaker());
        }
        return bandList;
    }


    public List<BandDto> GetDtoListFromBandList(IList<Band> bands)
    {
        var bandDtoList = bands.Select(GetDtoFromBand).ToList();
        return bandDtoList;
    }
}
